source/io/JavaScriptInvoke.cpp

```diff
@@ -43,13 +40,40 @@ VIREO_EXPORT const char* JavaScriptInvoke_GetParameterType(StaticTypeAndData *pa
         returnBuffer->Resize1D(0);
     }
 
+    return returnBuffer;
+}
+
+//------------------------------------------------------------
+//! Return parameter type name for the given element(index) in the parameters array
+VIREO_EXPORT const char* JavaScriptInvoke_GetParameterType(StaticTypeAndData *parameters, Int32 index)
+{
+    TypeRef parameterType = parameters[index]._paramType;
+    StringRef returnBuffer = AllocateReturnBuffer();
+
     if (returnBuffer) {
         SubString typeName = parameterType->Name();
         returnBuffer->Append(typeName.Length(), (Utf8Char*)typeName.Begin());
+
         // Add an explicit null terminator so it looks like a C string.
         returnBuffer->Append((Utf8Char)'\0');
         return (const char*)returnBuffer->Begin();
     }
+
+    return "";
```

Comment:
It might make more sense for this to return null. We don't really know the lifetime of that empty string once the function returns.